plugins/providers/gcloudiam/provider.go

```diff
@@ -50,6 +52,46 @@ func (p *Provider) GetResources(pc *domain.ProviderConfig) ([]*domain.Resource,
 		t = ResourceTypeOrganization
 	}
 
+	client, err := p.getIamClient(pc)
+	if err != nil {
+		return nil, err
+	}
+
+	iamRoles, err := client.GetRoles()
+	if err != nil {
+		return nil, err
+	}
+
+	var roles []*domain.Role
+	for _, r := range iamRoles {
+		roles = append(roles, &domain.Role{
+			ID:          r.Name,
+			Name:        r.Title,
+			Description: r.Description,
+		})
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	rolesMap := make(map[string]*domain.Role)
+	for _, role := range roles {
+		rolesMap[role.ID] = role
+	}
+
+	for _, rc := range pc.Resources {
+		for _, ro := range rc.Roles {
+			for _, p := range ro.Permissions {
+				permission := fmt.Sprint(p)
+				_, ok := rolesMap[permission]
+				if ok != true {
+					return nil, fmt.Errorf("%v: %v", ErrInvalidProjectRole, permission)
+				}
```

Comment:
```suggestion
				
				if _, ok := rolesMap[permission]; !ok {
					return nil, fmt.Errorf("%v: %v", ErrInvalidProjectRole, permission)
				}
```

Reply:
DOne

Reply:
Hi @utsav14nov @bsushmith , use `%w` to wrap error
```
fmt.Errorf("%w: %v", ErrInvalidProjectRole, permission)
```